lib/modules/manager/gradle-wrapper/utils.ts

```diff
@@ -61,6 +73,31 @@ export function getJavaConstraint(
   return '^11.0.0';
 }
 
+/**
+ * https://docs.gradle.org/current/userguide/gradle_daemon.html#sec:daemon_jvm_criteria
+ */
+export async function getJvmConfiguration(
+  gradlewFile: string,
+): Promise<string | null> {
+  const daemonJvmFile = join(
+    dirname(gradlewFile),
+    'gradle/gradle-daemon-jvm.properties',
+  );
+  const daemonJvm = await readLocalFile(daemonJvmFile, 'utf8');
+  if (daemonJvm) {
+    const TOOLCHAIN_VERSION_REGEX = regEx(
+      '^(?:toolchainVersion\\s*=\\s*)(?<version>\\d+)$',
+      'm',
+    );
```

Comment:
Can this be moved to the top level for performance reasons?

Reply:
Or rather, "at most once". If it's toplevel, it will be run exactly once, regardless if a daemon-jvm properties file exists or not.

Reply:
My thinking was that since `regEx()` has a `useCache` flag which is enabled by default, any re-use should be covered. This could happen e.g., if `gradle` and `gradle-wrapper` managers are enabled.